test/logic_value_test.dart

```diff
@@ -1022,4 +1022,201 @@ void main() {
+      final c = LogicValue.ofBigInt(BigInt.parse('-9223372036854775809'), 64);
+      final d = LogicValue.ofBigInt(BigInt.one, 64);
+      final e = LogicValue.ofInt(23, 64);
+      final f = LogicValue.ofInt(-4611686018427387903, 64);
+      final g = LogicValue.ofInt(32, 64);
+      final h = LogicValue.ofInt(-9223372036854775808, 64);
+
+      final values = <LogicValue>[a, b, c, d, e, f, g, h];
+
+      final expectedInt = List<int>.filled(values.length, 0);
+      final expectedBigInt = List<BigInt>.filled(values.length, BigInt.zero);
+      for (var i = 0; i < values.length; i++) {
+        expectedBigInt[i] = values[i].toBigInt();
+        expectedInt[i] = values[i].toInt();
+      }
+
+      values.sort();
+      expectedInt.sort();
+      expectedBigInt.sort();
+      for (var i = 0; i < values.length; i++) {
+        expect(values[i].toBigInt(), expectedBigInt[i]);
+        expect(values[i].toInt(), expectedInt[i]);
+      }
+    });
+
+    test('signed BigInt', () {
+      final a = LogicValue.ofBigInt(BigInt.parse('3'), 128);
+      final b = LogicValue.ofBigInt(BigInt.zero, 128);
+      final c = LogicValue.ofBigInt(BigInt.from(-1), 128);
+      final d = LogicValue.ofBigInt(BigInt.one, 128);
+      final e = LogicValue.ofBigInt(
+          BigInt.parse('340282366920938463463374607431768211455'), 128);
+      final f = LogicValue.ofBigInt(
+          BigInt.parse('-170141183460469231731687303715884105727'), 128);
+      final g = LogicValue.ofBigInt(
+          BigInt.parse('-170141183460469231731687303715884105728'), 128);
+      final h = LogicValue.ofBigInt(BigInt.one, 128);
+
+      final values = <LogicValue>[a, b, c, d, e, f, g, h];
+
+      final expected = List<BigInt>.filled(values.length, BigInt.zero);
+
+      for (var i = 0; i < values.length; i++) {
+        expected[i] = values[i].toBigInt();
+      }
+
+      values.sort();
+      expected.sort();
+      for (var i = 0; i < values.length; i++) {
+        expect(values[i].toBigInt(), expected[i]);
+      }
+    });
+    test('Exceptions', () {
+      final a64 = LogicValue.ofBigInt(BigInt.parse('3'), 64);
+      final a128 = LogicValue.ofBigInt(BigInt.parse('3'), 128);
+
+      final b64 = LogicValue.ofBigInt(BigInt.zero, 64);
+      final c64 = LogicValue.ofBigInt(BigInt.from(-1), 64);
+      final d64 = LogicValue.ofBigInt(BigInt.one, 64);
+      final e64 = LogicValue.ofInt(23, 64);
+      final e8 = LogicValue.ofInt(23, 8);
+      final f64 = LogicValue.ofInt(-9223372036854775808, 64);
+      final g64 = LogicValue.ofInt(-9223372036854775807, 64);
+      final h64 = LogicValue.ofInt(-1, 64);
+      final invalidLogicX = LogicValue.filled(8, LogicValue.x);
+      final invalidLogicZ = LogicValue.filled(8, LogicValue.z);
+
+      <LogicValue>[a64, b64, c64, d64, e64, f64, g64, h64].sort();
+
+      final excBigIntWidth = <LogicValue>[a64, b64, a128];
+      final excIntWidth = <LogicValue>[e64, e8, f64];
+
+      try {
```

Comment:
These try/catches are good if an exception is caught, but they dont test that the exception is actually being thrown.  If none of them threw any exception the test would still pass.